sway-core/src/parse_tree/literal.rs

```diff
@@ -22,6 +22,7 @@ pub enum Literal {
     U32(u32),
     U64(u64),
     String(span::Span),
+    Numeric(span::Span),
```

Comment:
Any reason not to store a numeric as a `u64`?

Reply:
The idea here is to not resolve to an integer until later. So a `Literal` of type `Numeric` remains a `span::Span` until it is resolved later on to an `UnsignedInteger`. I guess I could store a `u64` and then truncate later. Is that preferred?

Reply:
Changed to `u64`.